dashboard/agent.py

```diff
@@ -172,9 +172,29 @@ async def _check_parent():
             """Check if raylet is dead and fate-share if it is."""
             try:
                 curr_proc = psutil.Process()
+                parent_death_cnt = 0
                 while True:
                     parent = curr_proc.parent()
-                    if parent is None or parent.pid == 1 or self.ppid != parent.pid:
+                    parent_gone = parent is None
+                    init_assigned_for_parent = parent.pid == 1
+                    parent_changed = self.ppid != parent.pid
+                    if parent_gone or init_assigned_for_parent or parent_changed:
+                        parent_death_cnt += 1
+                        parent_death_threshold = 5
```

Comment:
Will make a constant

I think configurable env is not necessary. I don't even know if this will actually work. It is primarily for debugging this issue; https://github.com/ray-project/ray/issues/29412. We decide whether or not we want to make it configurable after we ask the user to repro